compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1190,7 +1190,7 @@ impl<'context> Elaborator<'context> {
         self.current_item = Some(DependencyId::Global(global_id));
         let let_stmt = global.stmt_def;
 
-        if !self.in_contract
+        if !self.module_id().module(self.def_maps).is_contract
```

Comment:
Maybe we should remove `self.in_contract` entirely if it is redundant?
I suppose there's the impl case where we override the value.

Reply:
Potentially, we can do this in a separate PR however.